### PR TITLE
Changed api call to DB retrieval 

### PR DESCRIPTION
**Pull request by rohitsngh27**
# Resolves
https://jira.rax.io/browse/SALUS-1014

# What
Changed api call to DB retrieval where possible

# How
Created a method to get from DB and convert it to DTO

## How to test
This can be tested via unit test

### CODE REVIEW

**Review by itzg — DISMISSED** [review]
Looks good, but probably good to wait on one more approval.

**Review by jjbuchan — COMMENTED** [review]
Since you've removed the ApiClient you can also remove the section in the `pom.xml` relating to that.

That and the usual `Optional` change will make this complete.

**Comment by jjbuchan** [comment]
I just realized you might also be able to remove some of the API Client dependencies in other PRs as well.  Anywhere an API Client is no longer used we should be able to clean up the pom.

**Comment by rohitsngh27** [comment]
> I just realized you might also be able to remove some of the API Client dependencies in other PRs as well. Anywhere an API Client is no longer used we should be able to clean up the pom.

There's a resourceApi request for **getResourcesWithLabels** which is being used, That's why i didn't remove it from the pom.xml

**Comment by jjbuchan** [comment]
There's two different client pieces to the pom.  In this case I initially meant we can remove this part that builds/exports the api client module.

```
      <plugin>
        <artifactId>maven-jar-plugin</artifactId>
        <version>3.0.2</version>
        <executions>
          <execution>
            <id>client-jar</id>
            <goals><goal>jar</goal></goals>
            <configuration>
              <classifier>client</classifier>
              <includes>
                <include>**/web/client/**</include>
                <include>**/web/model/**</include>
              </includes>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

but for the other PRs I do mean that we should remove dependencies wherever possible too.  Thanks for already checking that here.

File: pom.xml [hunk 1]
```diff
@@ -179,7 +179,6 @@
             <configuration>
               <classifier>client</classifier>
               <includes>
-                <include>**/web/client/**</include>
                 <include>**/web/model/**</include>
               </includes>
             </configuration>
```
Comment (itzg) [inline pom.xml:184]: I think @jjbuchan 's [comment](https://github.com/racker/salus-telemetry-agent-catalog-management/pull/43#issuecomment-689723219) was asking to remove this entire `client-jar` execution.

**Review by itzg — COMMENTED** [review]
Changes look ok other than possibly still needing pom change.

File: pom.xml (hunk 1 above)
Comment (rohitsngh27) [reply pom.xml:184]: I think we still need model jar for ResourceDTO object which is in resource management web/model package

Comment (jjbuchan) [reply pom.xml:184]: This block is not related to Resources.  This is the client jar that can be imported by other applications.  It was previously only used by ambassador but you should be removing that now.

**Review by jjbuchan — CHANGES_REQUESTED** [review]

**Review by jjbuchan — APPROVED** [review]

**Review by itzg — APPROVED** [review]

**Pull request merged**